### PR TITLE
Sourcery Starbot ⭐ refactored zhqu1148980644/hictools

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>master</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/hictools master
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: hictools/chrommatrix.py [hunk 1]
```diff
@@ -280,10 +280,9 @@ def oe(self,
 
         if sparse:
             return spoe
-        else:
-            dsoe = (spoe + spoe.T).toarray()
-            get_diag(dsoe)[:] /= 2
-            return dsoe
+        dsoe = (spoe + spoe.T).toarray()
+        get_diag(dsoe)[:] /= 2
+        return dsoe
```
Comment (SourceryAI) [inline hictools/chrommatrix.py:285]: Function `ChromMatrix.oe` refactored with the following changes:
- Remove unnecessary else after guard condition ([`remove-unnecessary-else`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-unnecessary-else))

File: hictools/chrommatrix.py [hunk 2]
```diff
@@ -317,6 +316,3 @@ def output_handler(raw_fn, args, kwargs, extra_kwargs, output):
 
 
 MethodWrapper.wrap_attr(ChromMatrix, filter_fn, MethodWrapper(output_handler=output_handler))
-
-if __name__ == "__main__":
-    pass
```
Comment (SourceryAI) [inline hictools/chrommatrix.py:322]: Lines `320-322` refactored with the following changes:
- Remove redundant conditional ([`remove-redundant-if`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-redundant-if))

File: hictools/compartment.py [hunk 3]
```diff
@@ -124,5 +124,4 @@ def __call__(self, method: str = "pca", numvecs: int = 3, sort: bool = True, ful
         return self.chrom_ma.handle_mask(vecs, full=full)
 
 
-if __name__ == '__main__':
-    pass
+pass
```
Comment (SourceryAI) [inline hictools/compartment.py:127]: Lines `127-128` refactored with the following changes:
- Remove redundant conditional ([`remove-redundant-if`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-redundant-if))

File: hictools/peaks.py [hunk 4]
```diff
@@ -97,9 +97,11 @@ def region_to_kernel(*regions) -> np.ndarray:
...
 
         def rect(x_start, x_len, y_start, y_len):
-            return set((i, j)
-                       for i in range(x_start, x_start + x_len)
-                       for j in range(y_start, y_start + y_len))
+            return {
+                (i, j)
+                for i in range(x_start, x_start + x_len)
+                for j in range(y_start, y_start + y_len)
+            }
```
Comment (SourceryAI) [inline hictools/peaks.py:104]: Function `HiccupsPeaksFinder.fetch_kernels.rect` refactored with the following changes:
- Replace list(), dict() or set() with comprehension ([`collection-builtin-to-comprehension`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#collection-builtin-to-comprehension))

File: hictools/peaks.py [hunk 5]
```diff
@@ -316,7 +318,7 @@ def enrich_mask(contact_array: np.ndarray,
 
         def away_gap_mask(indices, gap_mask, extend_width) -> np.ndarray:
             """Return mask of valid peaks away from gap regions."""
-            for i in range(extend_width):
+            for _ in range(extend_width):
```
Comment (SourceryAI) [inline hictools/peaks.py:321]: Function `HiccupsPeaksFinder.away_gap_mask` refactored with the following changes:
- Replace unused for index with underscore ([`for-index-underscore`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#for-index-underscore))

File: hictools/peaks.py [hunk 6]
```diff
@@ -333,34 +335,34 @@ def build_results(peaks_tuple: tuple, binsize: int = 1) -> pd.DataFrame:
         """Aggregate peak-infos into a pd.DataFrame object.
         """
         region_names = ['donut', 'horizontal', 'vertical', 'lower_left']
-        num_region = len(region_names)
```
Comment (SourceryAI) [inline hictools/peaks.py:336]: Function `HiccupsPeaksFinder.build_results` refactored with the following changes:
- Move assignments closer to their usage ([`move-assign`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#move-assign))
- Swap if/else branches ([`swap-if-else-branches`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#swap-if-else-branches))
- Remove unnecessary else after guard condition ([`remove-unnecessary-else`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-unnecessary-else))

File: hictools/peaks.py [hunk 7]
```diff
@@ -370,7 +372,3 @@ class ImagePeaksFinder(object):
     3: Design a statitical test method for the assessment pf  randomness and enrichment. -> filter points.
     """
     pass
-
-
-if __name__ == "__main__":
-    pass
```
Comment (SourceryAI) [inline hictools/peaks.py:376]: Lines `373-376` refactored with the following changes:
- Remove redundant conditional ([`remove-redundant-if`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-redundant-if))

File: hictools/reconstruct.py [hunk 8]
```diff
@@ -25,8 +25,7 @@ def pairwise_distances(x: Tensor) -> Tensor:
     x_norm = (x ** 2).sum(1).reshape(-1, 1)
     y_t = x.T
     y_norm = x_norm.reshape(1, -1)
-    dist = x_norm + y_norm - 2.0 * x @ y_t
-    return dist
+    return x_norm + y_norm - 2.0 * x @ y_t
```
Comment (SourceryAI) [inline hictools/reconstruct.py:28]: Function `pairwise_distances` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: hictools/reconstruct.py [hunk 9]
```diff
@@ -65,7 +64,7 @@ def train(self, batches=10, steps=100, optim=optim.SGD, **kwargs):
         optimizer = optim([self._pos], **kwargs)
         for b_idx in range(batches):
             optimizer.zero_grad()
-            for i in range(steps):
+            for _ in range(steps):
```
Comment (SourceryAI) [inline hictools/reconstruct.py:67]: Function `ReConstruct.train` refactored with the following changes:
- Replace unused for index with underscore ([`for-index-underscore`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#for-index-underscore))

File: hictools/reconstruct.py [hunk 10]
```diff
@@ -131,8 +130,7 @@ def d_exp(self):
     def loss(self):
         d_exp = self.d_exp
         d = pairwise_distances(self._pos)
-        cost = ((d - d_exp) ** 2).mean() / d_exp.max()
-        return cost
+        return ((d - d_exp) ** 2).mean() / d_exp.max()
```
Comment (SourceryAI) [inline hictools/reconstruct.py:133]: Function `MetricMDS.loss` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: hictools/tad.py [hunk 11]
```diff
@@ -34,12 +34,10 @@ def standard_di(up, down):
         up = up.sum(axis=1)
         down = down.sum(axis=1)
         expected = (up + down) / 2.0
-        di_array = (np.sign(down - up) *
+        return (np.sign(down - up) *
                     ((up - expected) ** 2 + (down - expected) ** 2)
                     / expected)
 
-        return di_array
```
Comment (SourceryAI) [inline hictools/tad.py:41]: Function `standard_di` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: hictools/tad.py [hunk 12]
```diff
@@ -50,11 +48,9 @@ def adap_di(up, down):
         mean_down = down.mean(axis=1)
         var_up = np.square(up - mean_up[:, None]).sum(axis=1)
         var_down = np.square(down - mean_down[:, None]).sum(axis=1)
-        di_array = ((mean_down - mean_up) /
+        return ((mean_down - mean_up) /
                     np.sqrt((var_up + var_down) / (window_size * (window_size - 1))))
 
-        return di_array
```
Comment (SourceryAI) [inline hictools/tad.py:56]: Function `adap_di` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: hictools/tad.py [hunk 13]
```diff
@@ -209,5 +205,4 @@ def fetch_sum(r1, r2, c1, c2):
     return ris.sum(axis=0) / num_nonzero, ris
 
 
-if __name__ == "__main__":
-    pass
+pass
```
Comment (SourceryAI) [inline hictools/tad.py:208]: Lines `212-213` refactored with the following changes:
- Remove redundant conditional ([`remove-redundant-if`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-redundant-if))

File: hictools/hgserver/cli.py [hunk 14]
```diff
@@ -181,7 +181,7 @@ def serve(ctx, store_uri, host, port, paths, workers):
             futures.append(fut)
 
             # serving
-            for i in range(1, workers + 1):
+            for _ in range(1, workers + 1):
```
Comment (SourceryAI) [inline hictools/hgserver/cli.py:184]: Function `serve` refactored with the following changes:
- Replace unused for index with underscore ([`for-index-underscore`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#for-index-underscore))

File: hictools/hgserver/server.py [hunk 15]
```diff
@@ -63,11 +63,9 @@ async def chromsizes():
     @app.get('/api/v1/tileset_info/', response_class=UJSONResponse)
     async def tileset_info(uuids: List[str] = Query(None, alias="d")):
         tilesets = await fetch_tilesets(uuid=uuids)
-        info = {}
         existed_uuids = set(uuids) & set(tilesets.keys())
 
-        for uuid in existed_uuids:
-            info[uuid] = TileSet.tileset_info(tilesets[uuid])
+        info = {uuid: TileSet.tileset_info(tilesets[uuid]) for uuid in existed_uuids}
```
Comment (SourceryAI) [inline hictools/hgserver/server.py:68]: Function `create_app.tileset_info` refactored with the following changes:
- Convert for loop into dictionary comprehension ([`dict-comprehension`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#dict-comprehension))

File: hictools/hgserver/store.py [hunk 16]
```diff
@@ -179,19 +179,17 @@ def tileset_info(cls, tileset_dict):
...
+            return cooler_tiles.tiles(tileset_dict['datafile'], tids)
         elif filetype == "bigwig":
             import clodius.tiles.bigwig as bigwig_tiles
-            data = bigwig_tiles.tiles(tileset_dict['datafile'], tids)
+            return bigwig_tiles.tiles(tileset_dict['datafile'], tids)
         else:
-            data = {'error': f"Unknown tileset filetype: {filetype}"}
-
-        return data
+            return {'error': f"Unknown tileset filetype: {filetype}"}
```
Comment (SourceryAI) [inline hictools/hgserver/store.py:192]: Function `TileSet.tiles` refactored with the following changes:
- Lift return into if ([`lift-return-into-if`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#lift-return-into-if))

File: hictools/utils/io.py [hunk 17]
```diff
@@ -57,15 +57,15 @@ def _popen(command: str, file: str, mode: str) -> subprocess.Popen:
         :return:
         """
 
-        text = not (True if (len(mode) == 2 and mode[1] == 'b') else False)
+        text = len(mode) != 2 or mode[1] != 'b'
```
Comment (SourceryAI) [inline hictools/utils/io.py:60]: Function `auto_open._popen` refactored with the following changes:
- Simplify boolean if expression ([`boolean-if-exp-identity`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#boolean-if-exp-identity))
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))
- Simplify logical expression using De Morgan identities ([`de-morgan`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#de-morgan))

File: hictools/utils/io.py [hunk 18]
```diff
@@ -131,10 +129,7 @@ def _create_stream(self):
             self._stream = open(self._file, self.mode)
             return self._stream
 
-        if self.mode[0] == 'w':
-            self._stream = self._pipe.stdin
-        else:
-            self._stream = self._pipe.stdout
+        self._stream = self._pipe.stdin if self.mode[0] == 'w' else self._pipe.stdout
```
Comment (SourceryAI) [inline hictools/utils/io.py:132]: Function `auto_open._create_stream` refactored with the following changes:
- Replace if statement with if expression ([`assign-if-exp`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#assign-if-exp))

File: hictools/utils/io.py [hunk 19]
```diff
@@ -244,12 +239,12 @@ def extract_cool(cool_path: str, sub_cool_path: str, chroms: Iterable, intra_onl
...
     def open_mcool(filename, mode='r') -> h5py.File:
-        if '::' in filename:
-            filename, grp_name = filename.split('::')
-            return h5py.File(filename, mode)[grp_name]
-        else:
+        if '::' not in filename:
             return h5py.File(filename, mode)
 
+        filename, grp_name = filename.split('::')
+        return h5py.File(filename, mode)[grp_name]
```
Comment (SourceryAI) [inline hictools/utils/io.py:246]: Function `extract_cool.open_mcool` refactored with the following changes:
- Swap if/else branches ([`swap-if-else-branches`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#swap-if-else-branches))
- Remove unnecessary else after guard condition ([`remove-unnecessary-else`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-unnecessary-else))

File: hictools/utils/numtools.py [hunk 20]
```diff
@@ -163,7 +163,7 @@ def get_decay(mat: Union[np.ndarray],
         res_li = [next(mean_gen) for offset in range(st, ed)]
         res = agg_fn(res_li)
         res = 0 if np.isnan(res) else res
-        for i in range(st, ed):
+        for _ in range(st, ed):
```
Comment (SourceryAI) [inline hictools/utils/numtools.py:166]: Function `get_decay` refactored with the following changes:
- Replace unused for index with underscore ([`for-index-underscore`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#for-index-underscore))

File: hictools/utils/numtools.py [hunk 21]
```diff
@@ -232,10 +232,7 @@ def _fill_slice(slice_, length):
     @staticmethod
     def _is_slices(slices):
         try:
-            for slice_ in slices:
-                if not isinstance(slice_, slice):
-                    return False
-            return True
+            return all(isinstance(slice_, slice) for slice_ in slices)
```
Comment (SourceryAI) [inline hictools/utils/numtools.py:235]: Function `SliceMixin._is_slices` refactored with the following changes:
- Use any() instead of for loop ([`use-any`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#use-any))
- Invert any/all to simplify comparisons ([`invert-any-all`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#invert-any-all))

File: hictools/utils/utils.py [hunk 22]
```diff
@@ -86,18 +86,18 @@ def in_params(arg):
...
             return NotImplemented
 
+        # copy doc string to argument's help
+        for arg in target.params:
+            if not isinstance(arg, click.Option):
+                continue
+            desc = in_params(arg)
+            if desc and (not arg.help):
+                arg.help = desc
+        return target
```
Comment (SourceryAI) [inline hictools/utils/utils.py:99]: Function `paste_doc.decorate` refactored with the following changes:
- Swap if/else branches ([`swap-if-else-branches`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#swap-if-else-branches))
- Remove unnecessary else after guard condition ([`remove-unnecessary-else`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-unnecessary-else))

File: hictools/utils/utils.py [hunk 23]
```diff
@@ -116,9 +116,7 @@ def get_logger(name: str = None) -> logging.Logger:
         name_ = call_frame.f_code.co_name
         name = f"{file_}:{name_}" if name != '<module>' else file_
 
-    log = logging.getLogger(name)
-
-    return log
+    return logging.getLogger(name)
```
Comment (SourceryAI) [inline hictools/utils/utils.py:119]: Function `get_logger` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: hictools/utils/utils.py [hunk 24]
```diff
@@ -179,5 +177,4 @@ def wrap_attr(cls, wrap_cls, predicate: Callable[[str, object], bool], wrapper:
                 setattr(wrap_cls, attr, wrapper(attr, attr_obj))
 
 
-if __name__ == "__main__":
-    pass
+pass
```
Comment (SourceryAI) [inline hictools/utils/utils.py:180]: Lines `182-183` refactored with the following changes:
- Remove redundant conditional ([`remove-redundant-if`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-redundant-if))
